task/control-plane-agent/src/update/sp.rs

```diff
@@ -523,11 +524,36 @@ impl AcceptingData {
 
         // Did we write the last block?
         if self.next_write_offset == sp_image_size {
-            match sp_task.finish_image_update() {
-                Ok(()) => (State::Complete, Ok(())),
-                Err(err) => {
-                    (State::Failed(err), Err(SpError::UpdateFailed(err as u32)))
+            // Confirm that the image written is targeting the same board as our
+            // current image.  If the current image doesn't have a caboose or a
+            // `BORD` key, then we'll accept anything, but the incoming image
+            // **must** have a valid caboose and `BORD`.
+            const BOARD_KEY: [u8; 4] = *b"BORD";
+            let ours = userlib::kipc::get_caboose()
+                .map(CabooseReader::new)
+                .and_then(|reader| reader.get(BOARD_KEY).ok());
```

Comment:
Two questions:

1. Can this fail in a way that makes us think we don't have a `BORD`, but actually we do and reading it at runtime failed? (I.e., should we be checking the kind of error we get here and only proceeding on specific ones?)
2. Would it make sense to read the caboose and stuff it into packrat? (Not suggesting this happen as part of this PR, but just a general question.)